tsdb/store.go

```diff
@@ -155,6 +155,12 @@ func (s *Store) loadShards() error {
 				return err
 			}
 			for _, sh := range shards {
+				// Check to see if there are any `.tmp` directories that were left over from failed shard snapshots
+				if sh.IsDir() && strings.HasSuffix(sh.Name(), ".tmp") {
```

Comment:
Can we move this to `Engine.cleanup` so that this kind of logic is kept closer together?